### PR TITLE
Call GetNCVersionStatus API to validate the datapath setup

### PR DESCRIPTION
**Pull request by ashvindeodhar**
This change calls GetNCVersionStatus API for NMAgent to validate if the data path is set up. If the data path is set up, the NC version with CNS matches with the NC version programmed by NMAgent. A conservative approach is taken to check if the data path is set up i.e. only if GetNCVersionStatus successfully returns NC version mismatch, it is treated as error. All the other cases including error in calling this API are considered success.
This is multitenancy only change.

**Reason for Change**:

**Issue Fixed**:

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests

**Notes**:

### CODE REVIEW

**Comment by csfmomo** [comment]
TestGetNetworkContainerByOrchestratorContext  and TestGetInterfaceForNetworkContainer unit test seems have a nil return for api_test, we could re-run those unit test locally to see whether its flaky.

File: cns/nmagentclient/nmagentclient.go [hunk 1]
```diff
@@ -10,9 +10,11 @@ import (
 )
 
 const (
-	WireserverIP = "168.63.129.16"
+	GetNetworkContainerVersionURLFmt = "http://%s/machine/plugins/?comp=nmagent&type=NetworkManagement/interfaces/%s/networkContainers/%s/version/authenticationToken/%s/api-version/1"
```
[the remark this answers is not in the document]
Comment (ashvindeodhar) [reply cns/nmagentclient/nmagentclient.go:13]: I didn't use / touch the one in imdsclient package. If we are not using the imdsclient package at all we can just remove that whole code as a part of separate PR.

File: cns/restserver/api.go [hunk 2]
```diff
@@ -1184,6 +1194,17 @@ func (service *HTTPRestService) publishNetworkContainer(w http.ResponseWriter, r
 				logger.Errorf("[Azure-CNS] %s", returnMessage)
 			}
 		}
+
+		// Store ncGetVersionURL needed for calling NMAgent to check if vfp programming is completed for the NC
+		primaryInterfaceIdentifier := getInterfaceIdFromCreateNetworkContainerURL(req.CreateNetworkContainerURL)
+		authToken := getAuthTokenFromCreateNetworkContainerURL(req.CreateNetworkContainerURL)
+		ncGetVersionURL := fmt.Sprintf(nmagentclient.GetNetworkContainerVersionURLFmt,
```
[the remark this answers is not in the document]
Comment (ashvindeodhar) [reply cns/restserver/api.go:1201]: same comment as above.

File: cns/restserver/api_test.go [hunk 3]
```diff
@@ -87,10 +95,25 @@ func getInterfaceInfo(w http.ResponseWriter, r *http.Request) {
...
+	}
+
+	if strings.Contains(r.RequestURI, "nc-nma-fail-version-mismatch") {
+		w.WriteHeader(http.StatusOK)
+		w.Write([]byte(`{"httpStatusCode":"200","networkContainerId":"nc-nma-fail-version-mismatch","version":"0"}`))
+	}
+
+	if strings.Contains(r.RequestURI, "nc-nma-fail-500") {
+		w.WriteHeader(http.StatusInternalServerError)
+		w.Write([]byte(`{"httpStatusCode":"200","networkContainerId":"nc-nma-fail-500","version":"0"}`))
```
[the remark this answers is not in the document]
Comment (ashvindeodhar) [reply cns/restserver/api_test.go:111]: yes

File: cns/restserver/util.go [hunk 4]
```diff
@@ -332,28 +332,48 @@ func (service *HTTPRestService) getNetworkContainerResponse(req cns.GetNetworkCo
...
+
+		} else if service.ChannelMode == cns.Managed {
+			// If the NC goal state doesn't exist in CNS running in managed mode, call DNC to retrieve the goal state
+			var (
+				dncEP     = service.GetOption(acn.OptPrivateEndpoint).(string)
+				infraVnet = service.GetOption(acn.OptInfrastructureNetworkID).(string)
+				nodeID    = service.GetOption(acn.OptNodeID).(string)
+			)
+
+			service.Unlock()
```
[the remark this answers is not in the document]
Comment (ashvindeodhar) [reply cns/restserver/util.go:369]: We unlock before making syncNodeStatus call which is a call to DNC over network. We may spend seconds waiting for the response so don;t want to hold the lock for that duration

**Pull request merged**